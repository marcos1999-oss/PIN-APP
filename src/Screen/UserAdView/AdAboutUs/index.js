import React from "react";
import {
  View,
  Text,
  Button,
  List,
  ListItem,
  Left,
  Right,
  Icon,
} from "native-base";
import { Linking, TouchableOpacity } from 'react-native';
import { isEqual, isEmpty, head, last, join, map, sortBy, filter } from 'lodash';
import styles from "./style";
import { DetailedFeatureList } from '../../../Components/Common'

export default class AdAboutUs extends React.Component {
  static getScheduleName(schedule) {
    const dayNames = {
      '0': 'Sun',
      '1': 'Mon',
      '2': 'Tue',
      '3': 'Wed',
      '4': 'Thu',
      '5': 'Fri',
      '6': 'Sat',
    };
    const weekDays = sortBy(schedule.weekDays, [(wd) => parseInt(wd)]);

    if (isEmpty(weekDays)) {
      return 'Daily';
    }

    if (weekDays.length === 1) {
      return dayNames[weekDays[0]];
    }

    // all days
    if (isEqual(weekDays, ['0', '1', '2', '3', '4', '5', '6'])) {
      return 'Daily';
    }

    if (isEqual(weekDays, ['0', '6'])) {
      return 'Weekends';
    }

    // if it's a sequence (eg: [3, 4, 5])
    if ((parseInt(last(weekDays)) - parseInt(head(weekDays))) === (weekDays.length - 1)) {
      return `${dayNames[head(weekDays)]} - ${dayNames[last(weekDays)]}`;
    }

    return join(map(weekDays, (wd) => dayNames[wd]), '/');
  }

  static getScheduleHours(schedule) {
    return `${schedule.starts} - ${schedule.ends}`;
  }

  renderSchedules(schedules) {
    return map(filter(schedules, (s) => !isEmpty(s.weekDays)), (schedule) => {
      const scheduleName = AdAboutUs.getScheduleName(schedule);
      const scheduleHours = AdAboutUs.getScheduleHours(schedule);

      return (
        <View style={styles.schedule} key={schedule.id}>
          <View style={styles.dateContainer}>
            <Text style={styles.date}>{scheduleName}</Text>
          </View>

          <View style={styles.timeContainer}>
            <Text style={styles.time}>{scheduleHours}</Text>
          </View>
        </View>
      );
    });
  }

  render() {
    let websiteURL = `${this.props.post.firm.website}`;
    websiteURL = (websiteURL.includes('http') || websiteURL.includes('https')) ? websiteURL : `https://www.${this.props.post.firm.website}`;
    
    return (
      <View style={styles.container}>
        <Text style={styles.title}>{this.props.post.firm.name}</Text>
        <Text style={styles.summary} numberOfLines={3}>
          {this.props.post.firm.about}
        </Text>

        <DetailedFeatureList
          features={this.props.post.firm.features}
          outerContainerStyle={styles.featuresTable}
          containerStyle={styles.featuresContainer}
          imageContainerStyle={styles.featureImageContainer}
          imageStyle={styles.featureImage}
          textContainerStyle={styles.featureTextContainer}
          textStyle={styles.featureText}
        />

        <Text style={styles.title}>Store Hours</Text>

        <View style={styles.scheduleContainer}>
          {this.renderSchedules(this.props.post.firm.schedules)}
        </View>

        <List style={styles.bottomList}>
          <ListItem style={{ ...styles.listItem, borderTopWidth: 0.5 }} onPress={() => { (this.props.post && this.props.post.firm && this.props.post.firm.website) && Linking.openURL(websiteURL) }}>
            <Left>
              <Text>{this.props.post.firm.name} website</Text>
            </Left>

            <Right>
              <Icon name="arrow-forward" />
            </Right>
          </ListItem>

          <ListItem
            style={styles.listItem}
            onPress={() => this.props.navigation.navigate("ReportListing")}
          >
            <Left>
              <Text>Report this listing</Text>
            </Left>

            <Right>
              <Icon name="arrow-forward" />
            </Right>
          </ListItem>
        </List>
      </View>
    );
  }
}
