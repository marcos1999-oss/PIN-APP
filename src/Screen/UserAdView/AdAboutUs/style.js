import { StyleSheet } from "react-native";
import {
  width,
  verticalScale,
  moderateScale,
  scale,
} from "../../../Utils/scaling";
import {responsiveHeight} from "../../../Utils/dimensions";

const blackText = "#6E6969";
const green = "#3FC94F";
export const purpleText = "#A338FF";

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
  },
  title: {
    marginBottom:responsiveHeight(1),
    color: blackText,
    fontSize: moderateScale(12),
    fontWeight: "bold",
  },
  summary: {
    color: blackText,
    fontSize: moderateScale(14),
    marginBottom: verticalScale(33),
  },
  featuresTable: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginBottom: verticalScale(18),
  },
  featuresContainer: {
    flexDirection: "row",
    flexGrow: 0,
    flexBasis: "33%",
    marginBottom: verticalScale(10),
  },
  featureImageContainer: {
    width: scale(15),
    height: verticalScale(15),
    justifyContent: "flex-start",
    alignItems: "center",
  },
  featureImage: {
    flex: 1,
    resizeMode: "contain",
  },
  featureTextContainer: {
    flex: 1,
    justifyContent: "flex-start",
  },
  featureText: {
    marginLeft: scale(12),
    fontSize: moderateScale(10),
    color: blackText,
  },
  scheduleContainer: {
    marginTop: verticalScale(10),
  },
  schedule: {
    flexDirection: "row",
    marginBottom: verticalScale(17),
  },
  dateContainer: {
    flex: 7,
    justifyContent: "flex-start",
  },
  date: {
    color: blackText,
    fontSize: moderateScale(14),
  },
  timeContainer: {
    flex: 3,
    justifyContent: "flex-start",
  },
  time: {
    color: blackText,
    fontSize: moderateScale(12),
    fontWeight: "bold",
  },
  bottomList: {
    position: "relative",
    width,
    left: scale(-20),
    marginBottom: verticalScale(30),
  },
  listItem: {
    marginLeft: 0,
    paddingLeft: scale(25),
  },
});

export default styles;
