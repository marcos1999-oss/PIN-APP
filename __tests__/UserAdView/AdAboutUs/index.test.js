/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import AdAboutUs from '../../../src/Screen/UserAdView/AdAboutUs';

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const post = {
    firm: {
      name: 'test',
      features: [
        {
          id: 1,
          icon: 'add',
        }
      ],
    }
  };
  const tree = renderer.create(<AdAboutUs post={post} />).toJSON();
  expect(tree).toMatchSnapshot();
});
