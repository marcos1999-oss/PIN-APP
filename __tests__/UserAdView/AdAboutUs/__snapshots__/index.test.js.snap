// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#6E6969",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 13.34,
        },
      ]
    }
    uppercase={false}
  >
    test
  </Text>
  <Text
    numberOfLines={3}
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#6E6969",
          "fontSize": 21,
          "marginBottom": 66,
        },
      ]
    }
    uppercase={false}
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginBottom": 36,
      }
    }
  >
    <View />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#6E6969",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 13.34,
        },
      ]
    }
    uppercase={false}
  >
    Store Hours
  </Text>
  <View
    style={
      Array [
        Object {},
        Object {
          "marginTop": 20,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {},
        Object {
          "left": -40,
          "marginBottom": 60,
          "position": "relative",
          "width": 750,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "borderTopWidth": 0.5,
              "marginLeft": 0,
              "paddingLeft": 50,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
          >
            test
             website
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.28,
              "paddingBottom": null,
              "paddingLeft": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": null,
                    "color": "#c9c8cd",
                    "fontSize": 22,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <View
        onPress={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "flexDirection": "row",
              "marginLeft": 16,
              "paddingBottom": 13,
              "paddingRight": 16,
              "paddingTop": 13,
            },
            Object {
              "marginLeft": 0,
              "paddingLeft": 50,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 16,
              }
            }
            uppercase={false}
          >
            Report this listing
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "alignSelf": "center",
              "flex": 0.28,
              "paddingBottom": null,
              "paddingLeft": null,
              "paddingRight": null,
              "paddingTop": null,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": null,
                    "color": "#c9c8cd",
                    "fontSize": 22,
                  },
                ],
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
